### ISSUE TITLE
`describe` command is key-ordering sensitive

### ISSUE THREAD

**Issue by srackham** [nushell/nushell#11011]
### Describe the bug

The `describe` command reports lists of identical records that differ only record key ordering as type `list<any>` instead type table. Records should be insensitive to key ordering.

### How to reproduce

The following test fails with the assertion error: `They are not equal. Left = 'table<name: string, rank: int>'. Right = 'list<any>'`:

```nushell
use std assert

#[test]
def key_value_ordering [] {
    let rs1 =  [
        {name: sam rank: 10}
        {name: bill rank: 15}
    ]
    let rs2 = [
        {rank: 10 name: sam}
        {name: bill rank: 15}
    ]
    let type1 = ($rs1|describe)
    let type2 = ($rs2|describe)

    assert equal ($type1) ($type2)
}
```

### Expected behavior

The above test should pass.

### Configuration

| key                | value                                                            |
| ------------------ | ---------------------------------------------------------------- |
| version            | 0.85.0                                                           |
| branch             | makepkg                                                          |
| commit_hash        | a6f62e05ae5b4e9ba4027fbfffd21025a898783e                         |
| build_os           | linux-x86_64                                                     |
| build_target       | x86_64-unknown-linux-gnu                                         |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13) (Arch Linux rust 1:1.72.1-1) |
| cargo_version      | cargo 1.72.1                                                     |
| build_time         | 2023-09-22 08:35:37 +00:00                                       |
| build_rust_channel | release                                                          |
| allocator          | mimalloc                                                         |
| features           | dataframe, default, extra, sqlite, trash, which, zip             |
| installed_plugins  |                                                                  |

**Comment by fdncred** [comment]
I'm not convinced the test should pass but the I'd expect the second record set to be `table<rank: int, name: string>`. 

Although both answers, table and list, are technically correct because tables are lists of records.

**Comment by amtoine** [comment]
yeah, i agree the test is not passing for the wrong reason :thinking: 
it shouldn't pass because `table<name: string, rank: int>` is different `table<rank: int, name: string>` and not because `list<any>` is different than the first table type!

**Comment by srackham** [comment]
My concern is that when records have different key ordering `describe` reports a `list` data type; when the keys have the same order it reports a `table` data type. This implies that key ordering has semantic significance. But this is contradicted by the equality operator which reports that `rs1` is equal to `rs2`:

```nushell
assert equal $rs1 $rs2  # PASS.
```

I think the current behaviour is confusing, it implies that semantically identical items are somehow not identical. As it stands, the difference in `describe` behaviour reflects nothing more than a lexical difference in the source code formatting.

**Comment by amtoine** [comment]
also another fun one :eyes:
```nushell
> let a = {a: foo, b: bar}
> let b = {b: bar, a: foo}
> $a == $b
true
> ($a | describe) == ($b | describe)
false
```

**Comment by WindSoilder** [comment]
To make it pass, I'm afraid we needs to change `describe`'s prototype.

Currently it returns a string, so it just does a string comparison

**Comment by amtoine** [comment]
@WindSoilder 
i made a few attempts at structuring the output of `describe`, i would love that :pray:

**Comment by ItsHarper** [comment]
> yeah, i agree the test is not passing for the wrong reason 🤔 it shouldn't pass because `table<name: string, rank: int>` is different `table<rank: int, name: string>` and not because `list<any>` is different than the first table type!

When I originally read this, I thought you hadn't noticed that the second table did not have consistent ordering within it.

I'm now realizing that you're saying that the second table should be typed using the same order as its first element (as you described in [this discussion](https://github.com/nushell/nushell/discussions/10301#discussioncomment-6960902)).

Now that I know that that's the intended way to consistently handle tables created from records with inconsistent ordering, I no longer have a strong opinion about whether the commented test should pass or not, but I wanted to provide some extra clarity for anyone else looking in.

The bug that results in the creation of a `list<any>` is more directly documented in #11646
